aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AuthConstants.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito
+
+object AuthConstants {
```

Comment:
Most of these constants are 1 time use variables. It may be beneficial to just use the string if it is single use. I don't see any constants being used here that are spanning across multiple classes, warranting a generic AuthConstants scope.

Reply:
SECRET_HASH and USERNAME are used in more than 1 place.

Reply:
Can let others weight in, but I'm not a big fan of generic Constants classes. Its not a huge deal though. I would still remove any Constants that aren't used more than once. For those that are (SECRET_HASH and USERNAME), looks like you could do something like this:

```
class AuthParamsMap(username: String, secretHash: String?) : MutableMap<String, String> by mutableMapOf() {
    init {
        put("USERNAME", username)
        secretHash?.let { put("SECRET_HASH", it) }
    }
}
```

@div5yesh @sktimalsina Thoughts?

Reply:
We can refactor this once we are done with GA if we do not see value in having it at that time.

Reply:
Constants class is a slippery slope where one might end up in duplicate constants with slightly different name etc. It also becomes difficult to refactor them. 
I recommend to fix it when we start writing it rather than for later by defining constants only within the class of where it is used.  For example, KEY_SALT, KEY_SRP_A etc does not need to be defined beyond `SRPCognitoActions`, no one needs to know about it. If we remove this class, those constants goes away with it with bonus of compiler time saved. 
For cases where USERNAME is used in 3 classes, I'd still recommend to define it in 3 classes. But this is my opinion.

Reply:
I'm fine with keeping the constants within the usage class. 

I don't recommend caching USERNAME and HASH in an object. It is okay to recreate params for each sign in action instead of tying them together with shared objects.